price-feeder/oracle/client/chain_height.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/rs/zerolog"
+	tmrpcclient "github.com/tendermint/tendermint/rpc/client"
+	tmctypes "github.com/tendermint/tendermint/rpc/core/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+var (
+	errParseEventDataNewBlockHeader = errors.New("error parsing EventDataNewBlockHeader")
+	queryEventNewBlockHeader        = tmtypes.QueryForEvent(tmtypes.EventNewBlockHeader)
+)
+
+// ChainHeight is used to cache the chain height of the
+// current node which is being updated each time the
+// node sends an event of EventNewBlockHeader.
+type ChainHeight struct {
+	Logger zerolog.Logger
+
+	mtx               sync.RWMutex
+	errGetChainHeight error
+	lastChainHeight   int64
+}
+
+var (
+	// keep the instance thread safe.
+	once sync.Once
+
+	// keeps the single instance of ChainHeight.
+	instance *ChainHeight
```

Comment:
```suggestion
	chainHeightCache *ChainHeight
```

Reply:
this variable is in a global scope, so let's make it more meaningful.

Reply:
Also, can we remove it from the global scope?

Reply:
we can, but the intention is to make it an singleton (single instance in the whole bin)

Reply:
But is this function called more than once in the bin?

Reply:
not right now... 
but this way, whatever other part on price-feeder needs the block height is going to get from the same instance

Reply:
OK, maybe we can keep it. But in general, using global variables is a bad pattern and usually leads to technical debt. So if there is a way to avoid it then let's do it.

Reply:
we can remove it, but if in another part of the code, that is being called, we should use an single instance

Reply:
[singleton](https://medium.com/golang-issue/how-singleton-pattern-works-with-golang-2fdd61cd5a7f) removed in https://github.com/umee-network/umee/pull/1069/commits/03a356f61401903971533335d75821a977cd2f1d